dbms/src/Storages/StorageHDFS.cpp

```diff
@@ -127,6 +132,49 @@ class HDFSBlockOutputStream : public IBlockOutputStream
     BlockOutputStreamPtr writer;
 };
 
+Strings LSWithRegexpMatching(const String & path_for_ls, const HDFSFSPtr & fs, const String & for_match)
```

Comment:
I think it is acceptable to have separate functions for HDFS and local file system, because if we had a single function, we'd have to convert file data to the same format, and also pass it a callback for listing subdirectories, and it would be even more verbose.